pkg/kv/kvserver/kvserverpb/log.go

```diff
@@ -10,6 +10,15 @@
 
 package kvserverpb
 
+import "github.com/cockroachdb/cockroach/pkg/clusterversion"
+
+func init() {
+	// In the real world, when the field below was deprecated, we'd have
```

Comment:
2¢: Eh, I donno if I agree with this comment's assertion that we should be linking to a named version?

Let's say that ReplicaState{}.DeprecatedUsingAppliedStateKey is still referenced by/used by 22.1, so we need to keep it as long as 22.1 is in the cluster, but can get rid of it once we're on 22.2+ binaries (i.e. the cluster verison is >22.1).

In that case, I'd `Refer` to to `clusterversion.V22_1`, since that's the maximum version where we still need the field, and when we delete that version, we can delete the field.